### PR TITLE
Fix #395 - Empty declaration for is(... == module), package, __vector + remove typedef

### LINKED ISSUE #395: Empty declaration for is(... == module)
As found in DRuntimes `src/core/internal/execinfo.d`:
```
enum bool hasExecinfo = is(_execinfo == module);
```

```
src/core/internal/execinfo.d(71:25)[error]: Declaration expected
src/core/internal/execinfo.d(71:48)[warn]: Empty declaration
```

### PR DESCRIPTION
**Pull request by MoonlightSentinel**

### CODE REVIEW

**Comment by MoonlightSentinel** [comment]
@WebFreak001 Should i include the other tokens you mentioned in https://github.com/dlang-community/libdparse/issues/395#issuecomment-621805576?

File: src/dparse/parser.d [hunk 1]
```diff
@@ -7015,6 +7015,7 @@ class Parser
         case tok!"union":
         case tok!"class":
         case tok!"interface":
+        case tok!"module":
```
Comment (WebFreak001) [inline src/dparse/parser.d:7018]: didn't update documentation

Comment (MoonlightSentinel) [reply src/dparse/parser.d]: Fixed

**Comment by MoonlightSentinel** [comment]
Added all missing specializations and removed `typedef`

**Comment by WebFreak001** [comment]
typedef was deprecated in 2.057 and was completely removed in 2.072, so it's reasonable to remove it now.

**Pull request merged**